### PR TITLE
Enhance wait command interface

### PR DESCRIPTION
**Pull request by JesterEE**
Added:
 * Measure (Sense) Wait Offset  
   * smu.sense.wait_offset
 * Source (Output) Wait Subsystem
   * smu.output.wait_auto
   * smu.output.wait_gain
   * smu.output.wait_offset
   * smu.output.wait_state

### CODE REVIEW

**Comment by t-onoz** [comment]
Dear JesterEE,

I appreciate your pull request.
As for Source Wait Subsystem, it might be better to set under smu.source rather than smu.output, following the original subsystem hierarchy of B290X.
Any thoughts?

**Comment by JesterEE** [comment]
@t-onoz 
I do think that is more appropriate.  The reason I implemented it in `class Output` is because I was getting errors on tab completion (Spyder) when working with the driver.
For example:
```py
>>> smu = slave.agilent.b2900.B2900(slave.transport.Visa('GPIB::23'))
>>> smu.source. #  <TAB>
```
<details><summary>Click to expand error</summary>

```py
>>> smu = slave.agilent.b2900.B2900(slave.transport.Visa('GPIB::23'))
>>> smu.source. #  <TAB>
ERROR:slave.protocol:Exception occured on 1. try. Msg: Timeout(VisaIOError('VI_ERROR_TMO (-1073807339): Timeout expired before operation completed.',),) Retrying.
Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\jedi\cache.py", line 119, in wrapper
    return dct[key]
KeyError: (('sweep_points',), frozenset())

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 217, in _wrap_visa_exceptions
    yield
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 308, in __read__
    return self._instrument.read_raw(num_bytes)
  File "C:\Anaconda3\lib\site-packages\pyvisa\resources\messagebased.py", line 306, in read_raw
    chunk, status = self.visalib.read(self.session, size)
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\functions.py", line 1582, in read
    ret = library.viRead(session, buffer, count, byref(return_count))
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\highlevel.py", line 188, in _return_handler
    raise errors.VisaIOError(ret_value)
pyvisa.errors.VisaIOError: VI_ERROR_TMO (-1073807339): Timeout expired before operation completed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\slave\protocol.py", line 56, in wrapped
    return fn(self, transport, *args, **kw)
  File "C:\Anaconda3\lib\site-packages\slave\protocol.py", line 155, in query
    response = transport.read_until(self.resp_term.encode(self.encoding))
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 100, in read_until
    self._buffer += self.__read__(self._max_bytes)
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 308, in __read__
    return self._instrument.read_raw(num_bytes)
  File "C:\Anaconda3\lib\contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 220, in _wrap_visa_exceptions
    raise_with_traceback(Visa.Timeout(e))
  File "C:\Anaconda3\lib\site-packages\future\utils\__init__.py", line 419, in raise_with_traceback
    raise exc.with_traceback(traceback)
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 217, in _wrap_visa_exceptions
    yield
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 308, in __read__
    return self._instrument.read_raw(num_bytes)
  File "C:\Anaconda3\lib\site-packages\pyvisa\resources\messagebased.py", line 306, in read_raw
    chunk, status = self.visalib.read(self.session, size)
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\functions.py", line 1582, in read
    ret = library.viRead(session, buffer, count, byref(return_count))
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\highlevel.py", line 188, in _return_handler
    raise errors.VisaIOError(ret_value)
slave.transport.Visa.Timeout: VI_ERROR_TMO (-1073807339): Timeout expired before operation completed.
ERROR:slave.protocol:Exception occured on 2. try. Msg: Timeout(VisaIOError('VI_ERROR_TMO (-1073807339): Timeout expired before operation completed.',),) Clearing device and retrying.
Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\jedi\cache.py", line 119, in wrapper
    return dct[key]
KeyError: (('sweep_points',), frozenset())

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 217, in _wrap_visa_exceptions
    yield
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 308, in __read__
    return self._instrument.read_raw(num_bytes)
  File "C:\Anaconda3\lib\site-packages\pyvisa\resources\messagebased.py", line 306, in read_raw
    chunk, status = self.visalib.read(self.session, size)
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\functions.py", line 1582, in read
    ret = library.viRead(session, buffer, count, byref(return_count))
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\highlevel.py", line 188, in _return_handler
    raise errors.VisaIOError(ret_value)
pyvisa.errors.VisaIOError: VI_ERROR_TMO (-1073807339): Timeout expired before operation completed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Anaconda3\lib\site-packages\slave\protocol.py", line 60, in wrapped
    return fn(self, transport, *args, **kw)
  File "C:\Anaconda3\lib\site-packages\slave\protocol.py", line 155, in query
    response = transport.read_until(self.resp_term.encode(self.encoding))
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 100, in read_until
    self._buffer += self.__read__(self._max_bytes)
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 308, in __read__
    return self._instrument.read_raw(num_bytes)
  File "C:\Anaconda3\lib\contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 220, in _wrap_visa_exceptions
    raise_with_traceback(Visa.Timeout(e))
  File "C:\Anaconda3\lib\site-packages\future\utils\__init__.py", line 419, in raise_with_traceback
    raise exc.with_traceback(traceback)
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 217, in _wrap_visa_exceptions
    yield
  File "C:\Anaconda3\lib\site-packages\slave\transport.py", line 308, in __read__
    return self._instrument.read_raw(num_bytes)
  File "C:\Anaconda3\lib\site-packages\pyvisa\resources\messagebased.py", line 306, in read_raw
    chunk, status = self.visalib.read(self.session, size)
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\functions.py", line 1582, in read
    ret = library.viRead(session, buffer, count, byref(return_count))
  File "C:\Anaconda3\lib\site-packages\pyvisa\ctwrapper\highlevel.py", line 188, in _return_handler
    raise errors.VisaIOError(ret_value)
slave.transport.Visa.Timeout: VI_ERROR_TMO (-1073807339): Timeout expired before operation completed.
```
</details><p>

I didn't want to get too deep into fixing the driver at the time (Sorry! Maybe I can later ... next week?) and this worked better for me for the time being ... and it was next to the filter commands so it was convenient. 

So yes to moving to the smu.source! 💯

**Comment by t-onoz** [comment]
@JesterEE 

I'm not sure about Spyder tab completion (I use PyCharm and Jupyter Qt Console and it works fine), but 
 I did find a bug in source.sweep_points which I fixed in the latest commit (https://github.com/t-onoz/slave/commit/92b8519b3e9bc2f025ea65d8818e19b6cacee3de).

Anyway I will merge your request later!

**Comment by JesterEE** [comment]
@t-onoz

I'll update, check it out in Spyder, and let you know if that was it.  Then I'll update the pull to have smu.source used for the wait commands.

**Comment by JesterEE** [comment]
@t-onoz 
I updated my package with your commit and am no longer getting the error when implementing a `B2900` object with the `slave.transport.SimulatedTransport` class in the Spyder IDE.  I don't have access to the instrument for a couple days, but as soon as I can test it with the actual equipment I will update this pull.

File: slave/agilent/b2900.py [hunk 1]
```diff
@@ -561,7 +585,7 @@ def __init__(self, transport, protocol):
         super().__init__(transport, protocol)
         self.sense_elements = Command(':FORM:ELEM:SENS?', ':FORM:ELEM:SENS', Stream(Mapping(self._functions)))
 
-
+
```
Comment (JesterEE) [inline slave/agilent/b2900.py]: Adding this was a mistake ... removing in another commit

**Pull request merged**